File: tests/js/math_algorithms/sieve_of_eratosthenes.js
// vybe-test: js/math_algorithms/sieve_of_eratosthenes
// origin: languages/js/tests/js/test_math_algorithms.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

function sieve(n) {
    const primes = Array(n + 1).fill(true);
    primes[0] = primes[1] = false;
    for (let i = 2; i * i <= n; i++) {
        if (primes[i]) for (let j = i*i; j <= n; j+=i) primes[j] = false;
    }
    return primes.map((p,i)=>p?i:-1).filter(n=>n>0);
}
const p = sieve(30);
console.log(p.join(","));

File: tests/c/algorithms/sieve_of_eratosthenes.c
// vybe-test: c/algorithms/sieve_of_eratosthenes
// origin: languages/c/tests/c/test_algorithms.rs
#include <stdio.h>
#include <string.h>
#include <assert.h>

int is_prime[20];
void sieve(int n) {
    for (int i = 0; i < n; i++) is_prime[i] = 1;
    is_prime[0] = is_prime[1] = 0;
    for (int i = 2; i*i < n; i++)
        if (is_prime[i])
            for (int j = i*i; j < n; j += i) is_prime[j] = 0;
}
int main() {
const char *__w[] = {"8\n"};
int __n = 1, __i = 0;
sieve(20);
int count = 0;
for (int i = 0; i < 20; i++) if (is_prime[i]) count++;
{ char __t[512]; snprintf(__t, sizeof(__t), "%d\n", count);
  if (__i >= __n || strcmp(__t, __w[__i]) != 0) { printf("FAIL at line %d: got [%s]\n", __i, __t); assert(0); } __i++; }
if (__i != __n) { printf("FAIL: %d line(s), wanted %d\n", __i, __n); assert(0); }
return 0;
}

